framework/src/modules/chain/modules/blocks/chain.js

```diff
@@ -567,48 +567,32 @@ __private.loadSecondLastBlockStep = function(secondLastBlockId, tx) {
 };
 
 /**
- * Reverts changes on confirmed columns of mem_account for one transaction
- * @param {Object} transaction - transaction to undo
- * @param {Object} oldLastBlock - secondLastBlock
+ * Reverts confirmed transactions due to block deletion
+ * @param {Object} blockToBeUndone - secondLastBlock
  * @param {Object} tx - database transaction
  */
-__private.undoConfirmedStep = function(transaction, oldLastBlock, tx) {
-	return new Promise((resolve, reject) => {
-		// Retrieve sender by public key
-		modules.accounts.getAccount(
-			{ publicKey: transaction.senderPublicKey },
-			(accountErr, sender) => {
-				if (accountErr) {
-					// Fatal error, memory tables will be inconsistent
-					library.logger.error(
-						'Failed to get account for undoing transaction to confirmed state',
-						accountErr
-					);
-					return setImmediate(reject, accountErr);
-				}
-				// Undoing confirmed transaction - refresh confirmed balance (see: logic.transaction.undo, logic.transfer.undo)
-				// WARNING: DB_WRITE
-				return modules.transactions.undoConfirmed(
-					transaction,
-					oldLastBlock,
-					sender,
-					undoConfirmedErr => {
-						if (undoConfirmedErr) {
-							// Fatal error, memory tables will be inconsistent
-							library.logger.error(
-								'Failed to undo transaction to confirmed state of account',
-								undoConfirmedErr
-							);
-							return setImmediate(reject, undoConfirmedErr);
-						}
-						return setImmediate(resolve);
-					},
-					tx
-				);
-			},
-			tx
-		);
-	});
+__private.undoConfirmedStep = async function(blockToBeUndone, tx) {
```

Comment:
Just use the variable name: `block`. It's unnecessary to name the variable `blockToBeUndone` based on my opinion.

Reply:
Done